client/components/data/purchases/site-purchases/index.jsx

```diff
@@ -1,65 +0,0 @@
-/**
- * External dependencies
- */
-import React from 'react';
-
-/**
- * Internal dependencies
- */
-import { fetchSitePurchases } from 'lib/upgrades/actions';
-import PurchasesStore from 'lib/purchases/store';
-import StoreConnection from 'components/data/store-connection';
-import sitesFactory from 'lib/sites-list';
-
-/**
- * Module variables
- */
-const sites = sitesFactory(),
-	stores = [
-		PurchasesStore,
-		sites
```

Comment:
For now we are subscribing to sites-list using the following pattern in all connected components (I agree this is an antipattern but this is the current one and I believe we should wait until sites-list is completly moved to its own reducer before refactoring):
```js
	componentWillMount() {
		debug( 'Mounting SiteSettings React component.' );
		this.props.sites.on( 'change', this.updateSite );
	}

	componentWillUnmount() {
		this.props.sites.off( 'change', this.updateSite );
	}

	updateSite() {
		this.setState( { site: this.props.sites.getSelectedSite() } );
	}
```
So on site change, we're calling `mapStateToProps` and updating the `sitePurchases` props here https://github.com/Automattic/wp-calypso/pull/6621/files#diff-6feabe612ef2a76186aa9cccfe2e50e7R237
I'm not sure what else you would like me to change here? :)